data/users/externalldap.py

```
@@ -594,123 +680,125 @@ def has_restricted_users(self) -> bool:
+                        except ldap.NO_SUCH_OBJECT as nsoe:
                             logger.debug(
-                                "Pagination is supported for this LDAP server but on last page"
+                                "NSO when trying to lookup results of search %s with filter %s: %s",
+                                user_search_dn,
+                                search_flt,
+                                str(nsoe),
+                            )
+                        except ldap.LDAPError as ldaperr:
+                            _log_ldap_error(
+                                ldaperr,
+                                additional=f"Error when trying to lookup results of search {user_search_dn} with filter {serch_flt}",
```

Ruff (0.14.7)

Undefined name `serch_flt`

(F821)